python/ray/data/grouped_dataset.py

```diff
@@ -144,6 +146,111 @@ def _aggregate_on(
         )
         return self.aggregate(*aggs)
 
+    def map_groups(
+        self,
+        fn: Union[CallableClass, Callable[[BatchType], BatchType]],
+        *,
+        compute: Optional[str] = None,
+        batch_format: str = "native",
+        **ray_remote_args,
+    ) -> "Dataset[Any]":
+        """Apply the given function to each group of records of this dataset.
+
+        While the map_groups() is very flexible, note that it comes with downsides:
+            - It may be slower than using more specific methods such as min(), max().
+            - It requires that each group fits in memory on a single node.
+        In general, prefer to use aggregate() methods instead of map_groups().
+
+        This is a blocking operation.
+
+        Examples:
+            >>> # Return a single record per group (list of multiple records in,
+            >>> # list of a single record out).
+            >>> ray.data.range(100).groupby(lambda x: x % 3).map_groups(
+            ...     lambda x: [min(x) / max(x)])
+
+            >>> # Return multiple records per group (dataframe in, dataframe out).
+            >>> df = pd.DataFrame(
+            ...     {"A": ["a", "a", "b"], "B": [1, 1, 3], "C": [4, 6, 5]}
+            ... )
+            >>> grouped = ray.data.from_pandas(df).groupby("A")
+            >>> grouped.map_groups(
+            ...     lambda g: g.apply(
+            ...         lambda c: c / g[c.name].sum() if c.name in ["B", "C"] else c
```

Comment:
Can we do something simpler here? For example, "g / g.mean()" # Normalize by mean of the group.

Reply:
"lambda g: g / g.mean()" is not a valid one for dataframe here: the column "A" is a str type so it cannot do division. 
If we make column "A" an int, then it will work (we can simply grouped.map_group(lambda g: g / g.sum())), but may be still not desirable, since we probably wouldn't want to normalize the column of group key.